src/main/java/com/linecorp/armeria/client/routing/ServerGroupRegistryHub.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A memory server group registry center.
+ */
+public class ServerGroupRegistryHub {
+
+    private static final Map<String, NodeSelector> serverGroups = new ConcurrentHashMap<>();
+
+    private static final Map<NodeSelectStrategy.SelectStrategyType, NodeSelectStrategy> selectStrategies = new ConcurrentHashMap<>();
+
+    /**
+     * Register server group.if the groupName has already registered and can't replace it,
+     * throw {@link ServerGroupRegisteredException}.
+     */
+    public static void registry(String groupName, ServerGroup serverGroup,
+                                NodeSelectStrategy.SelectStrategyType type) {
+        requireNonNull(groupName, "groupName");
+        requireNonNull(serverGroup, "serverGroup");
+        requireNonNull(type, "selectStrategyType");
+
+        if (serverGroups.containsKey(groupName)) {
+            ServerGroup registeredServerGroup = serverGroups.get(groupName).serverGroup();
+            if (!registeredServerGroup.isReplaceable()) {
+                throw new ServerGroupRegisteredException("Server group:" + groupName + " registered " +
+                        "already and can't replace.");
+            }
+        }
+
+        serverGroups.put(groupName, getNodeSelectStrategy(type).newSelector(serverGroup));
```

Comment:
Could you use `putIfAbsent()` so that the existing group is not replaced by accident?